apps/wallet/src/ui/app/hooks/useQredoAPI.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+
+import { useBackgroundClient } from './useBackgroundClient';
+import { useQredoInfo } from './useQredoInfo';
+import { QredoAPI } from '_src/shared/qredo-api';
+
+const API_INSTANCES: Record<string, QredoAPI> = {};
+
+export function useQredoAPI(qredoID?: string) {
+    const backgroundClient = useBackgroundClient();
+    const { data, isLoading, error } = useQredoInfo(qredoID);
+    const [api, setAPI] = useState(
+        () => (qredoID && API_INSTANCES[qredoID]) || null
+    );
+    useEffect(() => {
+        if (
+            data?.qredoInfo?.apiUrl &&
+            data?.qredoInfo?.accessToken &&
+            qredoID
+        ) {
+            const instance = API_INSTANCES[qredoID];
+            // if apiUrl changes that will mean the qredo ID will change
```

Comment:
I'm a little confused by this comment, we have two cases here: 1). the access token changed and 2). there's no API instance for this Qredo ID. I don't understand from reading how `apiUrl` changing fits in and why we don't need to do anything for it

Reply:
Yeah you are right, it was more like a note for me that when the apiURL changes which means we inject a new qredo token, the qredoID will change. Yeah it creates more confusion than providing extra context. I will remove it